### ISSUE TITLE
`unnecessary-ellipsis` false-positive when using `...` in lambda

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#6036]
### Bug description

Similarly to what's been reported in #5973, this:

```python
# pylint: disable=missing-module-docstring
print("x", lambda: ...)
```

results in a false-positive `unnecessary-ellipsis`.

While that might be a strange thing to do, I use it to make PyYAML able to save/restore `...`:

```python
# Translate ... to ellipsis in YAML.
YamlLoader.add_constructor('!ellipsis', lambda loader, node: ...)
YamlLoader.add_implicit_resolver('!ellipsis', re.compile(r'\.\.\.'), None)
```

I bisected this to 78eed6af303e0e9854e43528d17d2ada96f11c09, part of #5470 ("New checker - Detect use of unnecessary ellipsis").

### Command used

```shell
pylint repro2.py
```

### Pylint output

```shell
************* Module repro2
repro2.py:2:19: W2301: Unnecessary ellipsis constant (unnecessary-ellipsis)
```

### Expected behavior

No warning.

### Pylint version

```shell
pylint 2.13.2
astroid 2.11.2
Python 3.8.12 (default, Mar  3 2022, 18:19:28) 
[GCC 11.2.0]
```

(environment in which I originally encountered the bug - output above is from a git install of both astroid and pylint)

### OS / Environment

Archlinux
